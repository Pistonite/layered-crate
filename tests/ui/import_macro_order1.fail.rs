mod current {
    pub struct MyTypeInThisLayer;
    pub struct MyDepInThisLayer;
    pub mod crate_ {
        pub mod my_dep {
            pub struct MyType;
        }
        pub mod my_other_dep {
            pub struct MyOtherType;
        }
        pub mod my_more_dep {
            pub struct MyMoreType;
        }
    }
}

mod test {
    #[layered_crate::import]
    use current::{
        self::MyTypeInThisLayer,
        super::my_dep::MyType,
        super::my_other_dep::MyOtherType,
        super::my_more_dep::MyMoreType,
        self::MyDepInThisLayer,
    };
}


fn main() {}
