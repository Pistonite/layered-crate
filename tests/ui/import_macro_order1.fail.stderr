error: Please group self (current layer) imports and other imports separately for readability.
       Move this import up to be together with other `self` imports
  --> tests/ui/import_macro_order1.fail.rs:24:9
   |
24 |         self::MyDepInThisLayer,
   |         ^^^^
